x-pack/plugins/observability_solution/observability_ai_assistant/server/functions/context.ts

```diff
@@ -134,6 +136,18 @@ export function registerContextFunction({
           logger: resources.logger,
         });
 
+        analytics.reportEvent<RecallRanking>(RecallRankingEventType, {
```

Comment:
Can't we send this over with existing telemetry if the scores are part of the conversation? (We store the documents returned by ELSER in .data)

Reply:
Yes, that's what I asked in Slack, I can bake the same information into the conversation feedback (like you mentioned when we first spoke about this) but then we only collect this information if the user clicks to submit feedback.

If you think that's fine/preferred then I can easily change to do that instead, let me know!

Reply:
Are we only tracking conversations when the user submits feedback? I thought we captured it every time the user sends a message or something. Maybe @CoenWarmer can clarify?

Reply:
Yes, only when they click submit.

We **do** track each prompt they send but then only the user message, not the conversation.
I'm not sure what is easier to edit.

Reply:
One benefit of doing this explicitly (or whatever) during recall is that it'll cover both chat and contextual insights.

Reply:
@miltonhultgren's correct. We track a single user prompt when a user hits 'submit' in the Prompt Editor. We capture the whole conversation when a user presses on either the positive or negative feedback buttons.

Reply:
If we pass the scores as part of the conversation, it'll be a JSON string, not sure if that matters or not for how easy it'll be to evaluate later?

Reply:
I suppose we could move the telemetry gathering to [here](https://github.com/elastic/kibana/blob/2df44b9f7f76f3d03f6e32be7f2a39034f97c22e/x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_conversation.ts#L123-L126) for example if we want to capture the whole conversation on each prompt, I just don't know if that won't create a lot more data and then maybe the telemetry for chat feedback isn't needed anymore?

Reply:
Based on Slack convos, are we good to merge this as is, @dgieselaar ?